File: src/BalancedBinaryTree.java
public class BalancedBinaryTree {
    private static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
    }

    public static boolean isBalanced(TreeNode root) {
        return height(root) != -1;
    }

    private static int height(TreeNode root) {
        if (root == null) {
            return 0;
        }

        int leftDepth = height(root.left);
        int rightDepth = height(root.right);
        if (leftDepth == -1 || rightDepth == -1 || Math.abs(rightDepth - leftDepth) > 1) {
            return -1;
        }

        return 1 + Math.max(leftDepth, rightDepth);
    }
}

File: python/balanced_binary_tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def _is_balanced(self, root: TreeNode) -> int:
        if root is None:
            return 0

        left_height = self._is_balanced(root.left)
        right_height = self._is_balanced(root.right)
        if left_height == -1 or right_height == -1 or abs(left_height - right_height) > 1:
            return -1

        return 1 + max(left_height, right_height)

    def isBalanced(self, root: TreeNode) -> bool:
        if root is None:
            return True
        return self._is_balanced(root) is not -1
